docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md

```diff
@@ -0,0 +1,419 @@
+
+For other devices like RDMA and FPGA, the node has 1 RDMA and 1 FGPA, will report the following information:
+
+```yaml
+status:
+  capacity:
+    koordinator.sh/rdma: 100
+    koordinator.sh/fpga: 100
+  allocatable:
+    koordinator.sh/rdma: 100
+    koordinator.sh/fpga: 100
+```
+
+Why do we need `koordinator.sh/gpu-memory-ratio` and `koordinator.sh/gpu-memory` ? 
+When user apply 0.5/0.25 GPU, the user don't know the exact memory total bytes per GPU, only wants to use 
+half or quarter percentage of memory, so user can request the GPU memory with `koordinator.sh/gpu-memory-ratio`. 
+When scheduler assigned Pod on concrete node, scheduler will translate the `koordinator.sh/gpu-memory-ratio` to `koordinator.sh/gpu-memory` by the formulas:  ***allocatedMemory = totalMemoryOf(GPU)  * `koordinator.sh/gpu-memory-ratio`***, so that the GPU isolation can work.
+
+During the scheduling filter phase, the scheduler will do special processing for `koordinator.sh/gpu-memory` and `koordinator.sh/gpu-memory-ratio`. When a Pod specifies `koordinator.sh/gpu-memory-ratio`, the scheduler checks each GPU instance on each node for unallocated or remaining resources to ensure that the remaining memory on each GPU instance meets the ratio requirement.
+
+If the user knows exactly or can roughly estimate the specific memory consumption of the workload, he can apply for GPU memory through `koordinator.sh/gpu-memory`. All details can be seen below.
+
+Besides, when dimension's value > 100, means Pod need multi-devices. now only allow the value can be divided by 100.
+
+#### User apply device resources scenarios
+   
+##### Compatible with `nvidia.com/gpu`
+
+```yaml
+resources:
+  requests:
+    nvidia.com/gpu: "2"
+    cpu: "4"
+    memory: "8Gi"
+```
+
+The scheduler translates the `nvida.com/gpu: 2` to the following spec:
+
+```yaml
+resources:
+  requests:
+    koordinator.sh/gpu-core: "200"
+    koordinator.sh/gpu-memory-ratio: "200"
+    koordinator.sh/gpu-memory: "16Gi" # assume 8G memory in bytes per GPU
+    cpu: "4"
+    memory: "8Gi"
+```
+
+##### Apply whole resources of GPU or part resources of GPU
+
+```yaml
+resources:
+   requests:
+    koordinator.sh/gpu: "50"
+    cpu: "4"
+    memory: "8Gi"
+```
+
+The scheduler translates the `koordinator.sh/gpu: "50"` to the following spec:
+
+```yaml
+resources:
+  requests:
+    koordinator.sh/gpu-core: "50"
+    koordinator.sh/gpu-memory-ratio: "50"
+    koordinator.sh/gpu-memory: "4Gi" # assume 8G memory in bytes for the GPU
+    cpu: "4"
+    memory: "8Gi"
+```
+
+##### Apply `koordinator.sh/gpu-core` and `koordinator.sh/gpu-memory-ratio` separately
+
+```yaml
+resources:
+  requests:
+    koordinator.sh/gpu-core: "50"
+    koordinator.sh/gpu-memory-ratio: "60"
+    cpu: "4"
+    memory: "8Gi"
+```
+
+##### Apply `koordinator.sh/gpu-core` and `koordinator.sh/gpu-memory` separately
```

Comment:
if a container only requests `gpu-core` and `gpu-memory`, will the amount of `gpu-memory-ratio` resources for the node in the scheduler cache be incorrect? because `gpu-memory-ratio` resources may be not assumed.

Reply:
we will translate gpu-memory-ratio based on gpu-memory on concrete node.